src/main/java/org/veupathdb/service/eda/ds/plugin/pass/PieplotPlugin.java

```diff
@@ -0,0 +1,102 @@
+import static org.veupathdb.service.eda.ds.util.RServeClient.streamResult;
+import static org.veupathdb.service.eda.ds.util.RServeClient.useRConnectionWithRemoteFiles;
+
+public class PieplotPlugin extends AbstractPlugin<PieplotPostRequest, PieplotSpec> {
+
+  @Override
+  public String getDisplayName() {
+    return "Pie plot";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Visualize part-to-whole relationships across the categories of a variable";
+  }
+  
+  @Override
+  public List<String> getProjects() {
+    return List.of(CLINEPI_PROJECT, MICROBIOME_PROJECT);
+  }
+  
+  @Override
+  protected Class<PieplotSpec> getVisualizationSpecClass() {
+    return PieplotSpec.class;
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
+    return new ConstraintSpec()
+      .dependencyOrder("xAxisVariable", "facetVariable")
+      .pattern()
+        .element("xAxisVariable")
+          .maxValues(8)
+          .description("Variable must have 8 or fewer unique values.")
+        .element("facetVariable")
+          .required(false)
+          .maxVars(2)
+          .description("Variable(s) must be of the same or a parent entity of the main variable.")
+      .done();
+  }
+
+  @Override
+  protected void validateVisualizationSpec(PieplotSpec pluginSpec) throws ValidationException {
+    validateInputs(new DataElementSet()
+      .entity(pluginSpec.getOutputEntityId())
+      .var("xAxisVariable", pluginSpec.getXAxisVariable())
+      .var("facetVariable", pluginSpec.getFacetVariable()));
+  }
+
+  @Override
+  protected List<StreamSpec> getRequestedStreams(PieplotSpec pluginSpec) {
+    return ListBuilder.asList(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
+        .addVar(pluginSpec.getXAxisVariable())
+        .addVars(pluginSpec.getFacetVariable()));
+  }
+
+  @Override
+  protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    PieplotSpec spec = getPluginSpec();
+    String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "FALSE";
+
+    Map<String, VariableSpec> varMap = new HashMap<String, VariableSpec>();
+    varMap.put("xAxisVariable", spec.getXAxisVariable());
+    varMap.put("facetVariable1", getVariableSpecFromList(spec.getFacetVariable(), 0));
+    varMap.put("facetVariable2", getVariableSpecFromList(spec.getFacetVariable(), 1));
+      
+    useRConnectionWithRemoteFiles(dataStreams, connection -> {
+      connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
+          spec.getXAxisVariable(),
+          getVariableSpecFromList(spec.getFacetVariable(), 0),
+          getVariableSpecFromList(spec.getFacetVariable(), 1)));
+      connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
+      String cmd =
+          "plot.data::bar(" + DEFAULT_SINGLE_STREAM_NAME + ", map, " +
+              "'count', 'group', " +
```

Comment:
Are we planning to eventually allow the user to switch between 'count' and 'proportion'?

Reply:
I think I initially assumed we might and then decided that it probably wasn't very useful.. Its a plot meant to visually support part -to-whole relationships based on counts so it seemed like a pie in proportion mode would probably look the same but have less info cause it was no longer aware of the underlying counts. Am I crazy?

Reply:
No you're totally right it would look exactly the same just have less info :) In general i agree with you the proportion is not really useful. However, in the specific case of our specific audience, it seems they often like to see both? Thinking about the bar plot also - it has both proportion and count which, when no strata vars are selected, just changes the vertical axis.

Anyways, i'd be happy to only release "count" and wait for someone to ask for "proportion" explicitly

Reply:
I think I can just add it in case. I don't think it costs us any real work.